api/apis/job_handler.go

```diff
@@ -70,7 +70,7 @@ func (h *JobHandler) RegisterRoutes(router *mux.Router) {
 
 func parseJobGUID(jobGUID string) (string, string, bool) {
 	// Match job.type-GUID and capture the job type and GUID for later use
-	jobRegexp := regexp.MustCompile("([a-z_-]+[.][a-z_]+)-([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})")
+	jobRegexp := regexp.MustCompile("([a-z_-]+[.][a-z_]+)-([a-z]*-?[a-z]*-?[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})")
```

Comment:
This regex is becoming quite complex. I wonder if the new optional part for matching `cf-xxx-` could be made clearer by using a non-capturing group. E.g. `(?:cf-[a-z]+-)?` instead of `[a-z]*-?[a-z]*-?`. Or if there is a finite (for now) list of prefixes, it could be more explicit with `(?:cf-(?:org|space|route|proc)-)?` maybe?

Reply:
Updated to use the `(?:cf-[a-z]+-)?` format. It seems like that would be the best solution without having to add to the list of types if we add more.